---
title: "TCGA"
author: "Jeremy Leipzig"
date: "5/4/2016"
output: html_document
---

I am using normalized RNA-Seq gene expression counts from publicly available breast cancer data in the The Cancer Genome Atlas (TCGA).
The `TCGAbiolinks` Bioconductor package offers a handy mechanisms for getting this data into R dataframes, as well as deciphering sample barcodes.


```{r}
library(TCGAbiolinks)
library(MASS)
library(SummarizedExperiment)

query<-TCGAquery(tumor="brca",platform = "IlluminaHiSeq_RNASeqV2")

# do this once
# TCGAdownload(query, path = "brca", type = "rsem.genes.results")
# TCGAprepare(query,dir="brca",type="rsem.genes.results",save=TRUE,filename="brca.rds")

brca<-get(load("brca.rds"))

dataBRCA<-assay(brca,"raw_counts")

dataNorm <- TCGAanalyze_Normalization(tabDF = dataBRCA,geneInfo = geneInfo,method = "geneLength")

useData<-dataNorm

samplesN <- TCGAquery_SampleTypes(colnames(useData), typesample = c("NT","NB","NBC","NBM","NEBV"))
samplesT <- TCGAquery_SampleTypes(colnames(useData), typesample = c("TP","TR","TB","TRBM","TAP","TM","TAM","THOC","TBM"))

dataSamples<-useData[,c(samplesN,samplesT)]
tDataSamples<-as.data.frame(t(dataSamples))
tDataSamples$type<-as.vector(sapply(row.names(tDataSamples),function(x){if(x %in% samplesN){return("normal")}else{return("tumor")}}))

```

The breast cancer data set has `r nrow(useData)` (genes) and `r ncol(useData)` variables (samples). There are `r length(samplesN)` samples that are "N" (Normal) and `r length(samplesT)` are T (Tumor - either metastatic, primary, recurrent, etc.). These are all from different patients.

We can take a subset of genes useful in classifying breast cancer (PAM50), and then 50 genes chosen at random.

After a pivot and classifying the samples as tumor or normal, the data frame looks like this:

```{r}
pam50genesall<-row.names(pam50$centroids)
pam50genes<-pam50genesall[pam50genesall %in% colnames(tDataSamples)]
others <- colnames(tDataSamples)[sample(1:length(colnames(tDataSamples))-1, 50,replace=FALSE)]

pamsub<-tDataSamples[,c(pam50genes,others,"type")]
head(pamsub[,(length(colnames(pamsub))-10):length(colnames(pamsub))])
```

A linear discriminant analysis (aka Fisher) can be performed with bootstapped leave-one-out predictions.

```{r}
fit <- lda(formula = type ~ ., data = pamsub, CV=FALSE)

fitCV <- lda(formula = type ~ ., data = pamsub, CV=TRUE)

fit
```


### Goodness of Fit
A confusion matrix - predictions on the left, actuals on the top

```{r}
ct <- table(pamsub$type, fit$class)
ct
```

The total percent correct is `r sum(diag(prop.table(ct)))*100`

A test of proportions is show this is clearly better than expected
```{r}
prop.test(ct)
```





